Core/include/Acts/EventData/MultiTrajectory.hpp

```diff
@@ -467,10 +467,11 @@ class TrackStateProxy {
     std::shared_ptr<const Surface>& refSrf =
         m_traj->m_referenceSurfaces[dataref.irefsurface];
     // either unset, or the same, otherwise this is inconsistent assignment
-    assert(!refSrf || refSrf.get() == &meas.referenceSurface());
+    assert(!refSrf || refSrf.get() == meas.referenceSurface());
     if (!refSrf) {
       // ref surface is not set, set it now
-      refSrf = meas.referenceSurface().getSharedPtr();
+      refSrf = meas.referenceSurface()->getSharedPtr();
+      assert(refSrf != nullptr);
```

Comment:
Is that correct for shared pointer? I think you should use the Boolean operator overload.